### ISSUE TITLE
M1.2: Repo audits & security-wiring: stale numeric claims, HITL gate calibration, DetectorId audit, GUIDE integrity manifest

### ISSUE THREAD

**Issue by Mati86** [AI-Security-Internships-2026/02-soc-copilot-threat-analysis#30]
# M1.2: Repo audits & security-wiring: stale numeric claims, HITL gate calibration, DetectorId audit, GUIDE integrity manifest

## Issue Code: M1.2

---

## Concrete Tasks (PART A) — Stale Numeric Claims Audit

A1. Define stale-values set: `S = {0.7718, 0.7505, 0.7347, 3.616, p_value: 0.0}`.

A2. Run:
```bash
rg -n "0\.7718|0\.7505|0\.7347|3\.616|\"p_value\": 0\.0" . --glob '*.md' --glob '*.json' --glob '*.txt' --glob '*.py' -g '!Asma_IEEE/**' -g '!github-issue-drafts/**'
```
Write hits to `docs/stale_claims_audit.md` with filepath, lineno, OLD, ACTION, RATIONALE.

A3. For OLD=`0.7718`/`0.7505`/`0.7347` headlines outside `Asma_IEEE`: annotate inline `[STALE: train-sampled inflated figure; corrected held-out GUIDE_Test baseline = 0.6998]` — REPLACE only if clearly meant as held-out.

A4. For OLD=`3.616`: REPLACE with `measured via timeit (short-alert 1.93 µs, full-injection-alert 5.56 µs)`.

A5. For OLD=`p_value: 0.0`: REPLACE with `4.657e-12`.

A6. Re-run audit; any remaining hits must be inside intentional STALE annotations. Commit `docs/stale_claims_audit.md` + edited files.

---

## Concrete Tasks (PART B) — HITL Gate Calibration

B1. Read `src/agent/nodes.py` `review_decision()` and `src/agent/graph.py` HITL edge conditions. Locate every line where `confidence` or `llm_output["confidence"]` is read.

B2. Delete / comment-out every conditional branching on the LLM confidence string. Keep ONLY branches based on `state["ml_prediction_probs"][0] − state["ml_prediction_probs"][1]` (margin).

B3. Declare `T_init` named constant in `config/config.py` with docstring: `HITL auto-accept margin threshold; M5.1 sweep recommends 0.12 for 80/20 split`. Reconcile §4.6 current value (`0.20`) vs `0.12`; document decision.

B4. Write 5 pytest tests in `tests/test_hitl_gate_invariants.py`:
1. high-margin + LLM=`low` → accept
2. low-margin + LLM=`high` → escalate
3. swapped LLM conf identical alerts → identical decision (no causal effect)
4. `T=0.00` → 100% escalate
5. `T=1.00` → 0% accept

B5. Re-run M1.1 baseline-1 pytest 102 tests → confirm no regression.

---

## Concrete Tasks (PART C) — DetectorId + Feature Inclusion Audit

C1. Run large-sample schema check on `DetectorId` analogous to AlertTitle §3.2 inspection. Sample ≥50,000 unique `DetectorId` values from `GUIDE_train.csv`. Confirm all parse cleanly as int/float/string-numeric (no free text). Add assertion to `tests/test_schema_guardrail.py`: `validate_field_types("DetectorId", sample_50k) → all_pass`. If any non-numeric found → caveat or remove from numeric allowlist. Update paper Limitations note.

C2. Confirm whether `LastVerdict` and `SuspicionLevel` (target-adjacent leaky fields, Limitations flagged) are in the preprocessed feature matrix. Produce 1-paragraph decision memo: (a) keep + document as analyst-derived, (b) ablate + measure delta, or (c) document as leaky-but-known current state.

---

## Concrete Tasks (PART D) — Data Integrity Manifest

D1. On reproducible machine (M1.1 WSL2):
```bash
sha256sum datasets/GUIDE_train.csv datasets/GUIDE_Test.csv > datasets/INTEGRITY_MANIFEST.json
```
Format JSON with `sha256`, `file_size_bytes`, `last_modified_utc`.

D2. Edit `datasets/README.md` with 4 sections:
- (A) Dataset Source + Citation GUIDE (Ronen et al. 2024 arXiv)
- (B) Kaggle CLI install + download command
- (C) Microsoft alternative direct download
- (D) Integrity Check: run `python scripts/verify_data_integrity.py`

D3. Write `scripts/verify_data_integrity.py`:
1. load manifest
2. recompute SHA-256
3. match → exit 0 banner; mismatch → exit 1 with filename delta

D4. Run once; confirm exit 0.

---

## Acceptance Criteria

- [ ] (PART A) `docs/stale_claims_audit.md` committed with ≥1 entries (may be fewer if all staleness is paper-internal)
- [ ] (PART A) No remaining `3.616 µs` outside `Asma_IEEE/` + `github-issue-drafts/`; `p_value:0.0` replaced with `4.657e-12`; `0.7718`/`0.7347` all annotated or corrected
- [ ] (PART B) Static grep for `llm.*confidence` in routing conditionals → ZERO hits (display only)
- [ ] (PART B) `config/config.py` declares T constant with M5.1 sweep note + value-reconciliation rationale (`0.12` vs `0.20`)
- [ ] (PART B) `tests/test_hitl_gate_invariants.py` 5 new tests all pass; full 102 existing tests still pass
- [ ] (PART C) `DetectorId` 50K-sample audit completed; schema guardrail test; Limitations note updated with actual non-numeric % if any
- [ ] (PART C) `LastVerdict`/`SuspicionLevel` feature-inclusion audit + 1-paragraph decision memo committed
- [ ] (PART D) `datasets/INTEGRITY_MANIFEST.json` exists with 2 entries (`sha256`, size, mtime)
- [ ] (PART D) `datasets/README.md` contains citation + 2 download methods + integrity-check instructions
- [ ] (PART D) `scripts/verify_data_integrity.py` runs → exit 0 on current datasets
- [ ] M1.1 clean-environment Step 0 invokes `verify_data_integrity.py` → pass
- [ ] grep re-run (PART A) after edits → only intentional STALE annotations remain; all `3.616`/`p_value:0.0` replaced

**Comment by Mati86** [comment]
explicitly document why 0.20 is the selected operating threshold. We have to justify this selection in paper with strong argument.

**Comment by Mati86** [comment]
I will close this issue when you add a justification for selected threshold here.

**Comment by urpinklipbalm** [comment]
@Mati86 — threshold justification is now in the paper (Section 4.15), on PR #53 at `3a0a7ed`.

The strongest honest argument starts with a negative result, so the section leads with it.

## There is no Pareto knee

Marginal efficiency — accuracy gained on the auto-accepted subset per extra percentage point of analyst burden — is **flat between 0.23 and 0.38 across the entire sweep**, with no local maximum:

| T | burden | accepted acc | acc pts per burden pt |
|---|---|---|---|
| 0.02 | 2.0% | 0.7043 | 0.228 |
| 0.05 | 4.9% | 0.7129 | 0.299 |
| 0.10 | 9.3% | 0.7250 | 0.272 |
| 0.12 | 10.8% | 0.7302 | 0.344 |
| 0.15 | 13.6% | 0.7384 | 0.297 |
| **0.20** | **18.1%** | **0.7517** | **0.297** |
| 0.30 | 25.9% | 0.7799 | 0.359 |
| 0.50 | 43.2% | 0.8463 | 0.383 |

So the threshold is a **burden policy, not a hyperparameter with a correct value**. I'd rather say that plainly than present 0.20 as tuned — the data doesn't support the stronger claim. This also settles the 0.12 reconciliation `config.py` had deferred: nothing picks 0.12 over 0.20 on efficiency grounds.

## Given it's a policy choice, two independent anchors land on 0.20

**1 — the accept/escalate split.** 82.0/18.1 on the held-out 15,000, against 80.4/19.6 on the old 209-alert control. Two different populations, one forty times larger, agreeing within two points. 0.12 lands at 89/11.

**2 — a confidence-interval anchor, which picks 0.20 over 0.15 specifically.** The train-sampled figure this project used to report is **0.7357**.

- At **T=0.20**: accepted accuracy 0.7517, CI **[0.7440, 0.7592]** — entirely *above* 0.7357
- At **T=0.15**: accepted accuracy 0.7384, CI **[0.7310, 0.7463]** — still *contains* 0.7357

**0.20 is the smallest threshold at which the gate demonstrably delivers, on incidents the model has never seen, what the leaky evaluation used to claim.** That's the operational guarantee worth stating: the deployed config doesn't promise less in production than our own earlier numbers advertised.

## One caveat I've put in the paper rather than glossed

0.20 was fixed in Week 15 from the 209-alert sweep — **before** the held-out sweep existed. So M5.1 **confirms** the value, it does not derive it, and the paper says so. Had the held-out sweep disagreed, the honest outcome would have been to change the deployed value.

## Also

- `config/config.py` — the "revisit when M5.1 runs" note is resolved in place with the held-out table and these three findings. M5.1 has now run (#42).
- Two tests guard the **argument**, not the number: one fails if a different T ever becomes the smallest to clear 0.7357, the other fails if marginal efficiency stops being flat. Either would mean the framing has to change from "policy choice" to "tuned value" — it shouldn't be left standing silently.
- Source: `experiments/results/m5_1_burden_sweep.json`, swept over the class-balanced held-out 15,000 (0% incident overlap) — deliberately not a train-sampled set, since an operating point tuned on leaked data wouldn't hold in deployment.

Over to you for the close.
